pkg/clusteragent/admission/mutate/autoinstrumentation/config.go

```diff
@@ -165,6 +181,15 @@ type InstrumentationConfig struct {
 	// used. If no target matches, the auto instrumentation will not be applied. Full config key:
 	// apm_config.instrumentation.targets
 	Targets []Target `mapstructure:"targets" json:"targets"`
+	// ExcludeContainers configuration for containers to exclude from auto instrumentation.
+	// By default, istio-proxy containers are excluded, but this can be overwritten.
+	ExcludeContainers *ExcludeContainers `mapstructure:"exclude_containers" json:"exclude_containers"`
```

Comment:
Any reason not to make `ExcludeContainers []string`? Is there value for adding a custom struct type?

Reply:
The thing I was thinking about here was that if we ever wanted to add _anything_ else it would make it difficult to do it. Like container image names, or patterns, or anything else.

Reply:
Dropped configuration / UX for this.